java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -157,6 +188,48 @@ private static boolean generatesPrivateAccess(List<SymbolMetadata.AnnotationValu
     return values.stream().anyMatch(av -> "access".equals(av.name()) && "PRIVATE".equals(getAccessLevelValue(av.value())));
   }
 
+  private static boolean generatesPrivateAccess(ClassTree tree) {
+    if (usesAnnotation(tree, Collections.singletonList(LOMBOK_VALUE))) {
+      return true;
+    }
+    return Optional.ofNullable(tree.symbol().metadata().valuesForAnnotation(LOMBOK_FIELD_DEFAULTS))
+      .map(List::stream)
+      .map(s -> s.filter(Objects::nonNull)
+        .anyMatch(av -> "level".equals(av.name()) && "PRIVATE".equals(getAccessLevelValue(av.value())))
+      ).orElse(false);
+  }
+
+  private static boolean generatesPrivateAccess(VariableTree tree) {
+    Tree parent = tree.parent();
```

Comment:
I'm still not convinced about the logic here. For all field, you will look at the value of the annotation again, you should do it only once when you visit the class!